### ISSUE TITLE
session_start(): Failed to read session data: user (path: )

### ISSUE THREAD

**Issue by ZhangRuiMingZRM** [phalcon/cphalcon#14314]
* Phalcon version: `4.0.0`
* PHP Version: `7.3.6`
* Operating System: ubuntu
* Installation type: Compiling from source
* Server: php web server

```php
use Phalcon\Session\Manager;
use Phalcon\Session\Adapter\Redis;
use Phalcon\Storage\AdapterFactory;

try {
    $session = new Manager();
    $adapterFactory = new AdapterFactory();
    $handler = new Redis(
        $adapterFactory,
        [
            "host" => "127.0.0.1",
            "port" => 6379
        ]
    );
    $session->setHandler($handler);
    $session->start();
} 
...
```
I got this

```
PHP Warning:  session_start(): Failed to read session data: user (path: ) in .....
```

That is because [`SessionHandlerInterface::read`](https://www.php.net/manual/en/sessionhandlerinterface.read.php) expect `string` as return value but actually return `null` 

https://github.com/phalcon/cphalcon/blob/3d76cdaee5898d9566f59b8996c4ddbe4b29f729/phalcon/Session/Adapter/AbstractAdapter.zep#L54-L57

 `Phalcon\Session\Adapter\Redis::get()` may return `null`

https://github.com/phalcon/cphalcon/blob/3d76cdaee5898d9566f59b8996c4ddbe4b29f729/phalcon/Storage/Adapter/Redis.zep#L96-L102

**This is a solution** 

in `Phalcon/Session/Adapter/AbstractAdapter::read()`
```
    public function read(var id) -> string
    {
      var data;
        let data = this->adapter->get(id);

        return is_null(data) ? "" : data;
    }
```

And in test unit  `Phalcon\Test\Integration\Session\Adapter\Redis\ReadCest::sessionAdapterRedisRead()`  
should add test for another possible situation (if session_id(key) does not exist in redis)

https://github.com/phalcon/cphalcon/blob/3d76cdaee5898d9566f59b8996c4ddbe4b29f729/tests/integration/Session/Adapter/Redis/ReadCest.php#L36-L48

Test if session_id(key) does not exist in redis
```php
public function sessionAdapterRedisRead(IntegrationTester $I)
{
    .
    .
    . 
   $I->sendCommandToRedis('del', 'sess-reds-test1');
   $I->assertNotNull(
           $adapter->read('test1')
   );
}
```
So do as `Phalcon\Session\Adapter\Libmemcached::get()`

**Comment by ruudboon** [comment]
Thank for reporting. Do you have time to open a pull for this? Otherwise I can do this.

**Comment by ZhangRuiMingZRM** [comment]
> Thank for reporting. Do you have time to open a pull for this? Otherwise I can do this.

Very pleased to do it

**Comment by ruudboon** [comment]
Thnx!

**Comment by sergeyklay** [comment]
Fixed in the `4.0.x` branch. Feel free to open a new issue if the problem appears again. Thank you for the bug report.
